src/set_theory/cardinal/basic.lean

```diff
@@ -475,7 +478,7 @@ instance : canonically_ordered_comm_semiring cardinal.{u} :=
   le_self_add := λ a b, (add_zero a).ge.trans $ add_le_add_left (cardinal.zero_le _) _,
   eq_zero_or_eq_zero_of_mul_eq_zero := λ a b, induction_on₂ a b $ λ α β,
     by simpa only [mul_def, mk_eq_zero_iff, is_empty_prod] using id,
-  ..cardinal.comm_semiring, ..cardinal.linear_order }
+  ..cardinal.comm_semiring, ..cardinal.partial_order }
 
 instance : canonically_linear_ordered_add_monoid cardinal.{u} :=
```

Comment:
I think you need a computable `canonically_ordered_add_monoid` instance too.

Reply:
This is meant to address https://github.com/leanprover-community/mathlib4/pull/3247#discussion_r1161278929. I believe the computable instance we want is rather `comm_monoid`.

Reply:
Sure, but for every noncomputable isntance you add, you need to add an instance for every one of its computable parents. This applies for the `linear_ordered_comm_monoid_with_zero` instance too; so you need a `comm_monoid_with_zero` instance, and a `canonically_ordered_add_monoid` instance.

Reply:
Did I add them correctly? It's hard to tell since the whole file is noncomputable.

Reply:
I think so, though you can check with `#print` to see if they're computable.

Can you reorder them so that the computably additive instance is right before the corresponding non-computable one and same for the multiplicative one?